cranelift/codegen/meta/src/shared/instructions.rs

```diff
@@ -1129,25 +1129,29 @@ pub(crate) fn define(
 
     let H = &Operand::new("H", &entities.heap);
     let p = &Operand::new("p", HeapOffset);
-    let Size = &Operand::new("Size", &imm.uimm32).with_doc("Size in bytes");
+    let Offset = &Operand::new("Offset", &imm.uimm32).with_doc("Offset in bytes");
+    let Size = &Operand::new("Size", &imm.uimm8).with_doc("Size in bytes");
 
     ig.push(
         Inst::new(
             "heap_addr",
             r#"
-        Bounds check and compute absolute address of heap memory.
+        Bounds check and compute absolute address of ``p + Offset`` in heap memory.
 
-        Verify that the offset range ``p .. p + Size - 1`` is in bounds for the
+        Verify that the offset range ``p .. p + Offset + Size - 1`` is in bounds for the
```

Comment:
Tiny nit (and pre-existing) but worth trying to get right here: `..` in Rust usually means inclusive start, exclusive end, and with those semantics we should write this `p .. p + Offset + Size`. Or alternately use `..=`.

The plain meaning here might be clear enough to a "common-sense reader" but perhaps something like "offset range `p .. p + Offset + Size` (i.e., `Offset + Size` bytes starting at `p`)" would be even clearer?